src/games/behavmixed.imp

```diff
@@ -620,153 +533,135 @@ T MixedBehaviorProfile<T>::DiffNodeValue(const GameNode &p_node,
       // We've encountered the action; since we assume perfect recall,
       // we won't encounter it again, and the downtree value must
       // be the same.
-      return m_nodeValues(p_node->GetChild(p_oppAction->GetNumber())->GetNumber(),
-			  p_player->GetNumber());
+      return map_nodeValues[p_node->GetChild(p_oppAction)][p_player];
     }
     else {
-      T deriv = (T) 0;
-      for (int act = 1; act <= infoset->NumActions(); act++) {
-	deriv += (DiffNodeValue(p_node->GetChild(act), p_player, p_oppAction) *
-		  GetActionProb(infoset->GetAction(act)));
+      T deriv = T(0);
+      for (auto action : infoset->GetActions()) {
+	deriv += (DiffNodeValue(p_node->GetChild(action), p_player, p_oppAction) *
+		  GetActionProb(action));
       }
       return deriv;
     }
   }
   else {
     // If we reach a terminal node and haven't encountered p_oppAction,
     // derivative wrt this path is zero.
-    return (T) 0;
+    return T(0);
   }
 }
 
 //========================================================================
 //             MixedBehaviorProfile<T>: Cached profile information
 //========================================================================
 
+
+// compute realization probabilities for nodes and isets.
+template <class T>
+void MixedBehaviorProfile<T>::ComputePass1_realizProbs(const GameNode &node) const
```

Comment:
Would it be a bit more straightforward to pass in the realisation prob as an argument to the function when calling recursively?

Reply:
Left for now.